mac: NumPy version causes error message check to fail

This is not fully confirmed, but it's a pretty strong hunch; from Slack:

> [jwnimmer-tri](https://app.slack.com/team/U24GLHA7L) [Today at 5:56 PM](https://drakedevelopers.slack.com/archives/C270MN28G/p1589493403140300)
> 
> <https://drake-jenkins.csail.mit.edu/view/Production/job/mac-mojave-clang-bazel-continuous-release/723/> `FAIL: //bindings/pydrake:py/autodiffutils_test` (edited) 
> 
> 5 replies
> 
> * * * * *
> 
> 
> 
> [jwnimmer-tri](https://app.slack.com/team/U24GLHA7L)  [11 minutes ago](https://drakedevelopers.slack.com/archives/C270MN28G/p1589493410140400?thread_ts=1589493403.140300&cid=C270MN28G)
> ```
> (see /Users/mojave/workspace/mac-mojave-clang-bazel-continuous-release/_bazel_mojave/0623f93b365c16dbae8a9ca98583e26c/execroot/drake/bazel-out/darwin-opt/testlogs/bindings/pydrake/py/autodiffutils_test/test.log)\
> [4:19:57 PM]  INFO: From Testing //bindings/pydrake:py/autodiffutils_test:\
> [4:19:57 PM]  ==================== Test output for //bindings/pydrake:py/autodiffutils_test:\
> [4:19:57 PM]  .....F\
> [4:19:57 PM]  ======================================================================\
> [4:19:57 PM]  FAIL: test_scalar_api (autodiffutils_test.TestAutoDiffXd)\
> [4:19:57 PM]  ----------------------------------------------------------------------\
> [4:19:57 PM]  Traceback (most recent call last):\
> [4:19:57 PM]    File "/Users/mojave/workspace/mac-mojave-clang-bazel-continuous-release/_bazel_mojave/0623f93b365c16dbae8a9ca98583e26c/sandbox/darwin-sandbox/3751/execroot/drake/bazel-out/darwin-opt/bin/bindings/pydrake/py/autodiffutils_test.runfiles/drake/bindings/pydrake/test/autodiffutils_test.py", line 70, in test_scalar_api\
> [4:19:57 PM]      self.assertEqual(\
> [4:19:57 PM]  AssertionError: "don't know how to convert scalar number to float" != "float() argument must be a string or a nu[40 chars]fXd'"\
> [4:19:57 PM]  - don't know how to convert scalar number to float\
> [4:19:57 PM]  + float() argument must be a string or a number, not 'pydrake.autodiffutils.AutoDiffXd'\
> [4:19:57 PM]\
> [4:19:57 PM]\
> [4:19:57 PM]  ----------------------------------------------------------------------\
> [4:19:57 PM]  Ran 6 tests in 0.076s\
> [4:19:57 PM]\
> [4:19:57 PM]  FAILED (failures=1)\
> [4:19:57 PM]  ================================================================================
> ```
> 
> 
> [jwnimmer-tri](https://app.slack.com/team/U24GLHA7L)  [10 minutes ago](https://drakedevelopers.slack.com/archives/C270MN28G/p1589493459140700?thread_ts=1589493403.140300&cid=C270MN28G)
> 
> Changesets indicate <https://github.com/RobotLocomotion/drake/pull/13234> most likely
> 
> 
> 
> [jwnimmer-tri](https://app.slack.com/team/U24GLHA7L)  [10 minutes ago](https://drakedevelopers.slack.com/archives/C270MN28G/p1589493463140900?thread_ts=1589493403.140300&cid=C270MN28G)
> 
> \CC [@eric.cousineau](https://drakedevelopers.slack.com/team/U4PJQLBGR) [@sammy-tri](https://drakedevelopers.slack.com/team/U2AKZAAMR)
> 
> 
> 
> [eric.cousineau](https://app.slack.com/team/U4PJQLBGR)  [4 minutes ago](https://drakedevelopers.slack.com/archives/C270MN28G/p1589493837141100?thread_ts=1589493403.140300&cid=C270MN28G)
> 
> Hm... that stack trace just looks like another "NumPy changed out from under us on Homebrew" issue; I don't think MathematicalProgram is involved at all?
>

--- a/bindings/pydrake/test/autodiffutils_test.py
+++ b/bindings/pydrake/test/autodiffutils_test.py
@@ -67,9 +67,15 @@ class TestAutoDiffXd(unittest.TestCase):
         a_scalar = np.array(a)
         with self.assertRaises(TypeError) as cm:
             float(a_scalar)
-        self.assertEqual(
-            "don't know how to convert scalar number to float",
-            str(cm.exception))
+        if np.lib.NumpyVersion(np.__version__) < "1.14.0":
+            self.assertEqual(
+                "don't know how to convert scalar number to float",
+                str(cm.exception))
+        else:
+            self.assertEqual(
+                "float() argument must be a string or a number, not "
+                "'pydrake.autodiffutils.AutoDiffXd'",
+                str(cm.exception))
         # Test multi-element pass-through.
         x = np.array([AD(1.), AD(2.), AD(3.)])
         numpy_compare.assert_equal(autodiff_vector_pass_through(x), x)

